conn_query.go

```diff
@@ -99,3 +102,31 @@ func (c *connect) queryRow(ctx context.Context, release func(*connect, error), q
 		rows: rows,
 	}
 }
+
+var hasQueryParamsRe = regexp.MustCompile("{.+:.+}")
+
+func bindQueryOrAppendParameters(paramsProtocolSupport bool, options *QueryOptions, query string, timezone *time.Location, args ...interface{}) (string, error) {
+	// prefer native query parameters over legacy bind if query parameters provided explicit
+	if len(options.parameters) > 0 {
+		return query, nil
+	}
+
+	// validate if query contains a {<name>:<data type>} syntax, so it's intentional use of query parameters
+	// parameter values will be loaded from `args ...interface{}` for compatibility
+	if paramsProtocolSupport &&
+		len(args) > 0 &&
+		hasQueryParamsRe.MatchString(query) {
+		options.parameters = make(Parameters, len(args))
+		for _, a := range args {
```

Comment:
What happens if there is a mismatch here?  Should we raise an err?

Reply:
what mismatch exactly do you mean?

Reply:
Maybe I'm misunderstanding this, but if the arguments passed aren't a list of driver.NamedValues it just silently fails to add the parameters.  It's a minor point.

Reply:
I think it's a valid point. I was confused since your comment is referred by GH to another line, not to the type-casting one.

Since the user uses a `{xx:xx}` syntax explicitly we might let them know other arg types are not supported.